ige/include/ige/plugin/input/InputManager.hpp

```diff
@@ -33,10 +30,19 @@ class InputManager {
     Mouse& mouse();
     const Mouse& mouse() const;
 
+    Controller& add_controller(const ControllerId& id);
+
+    std::vector<Controller*> controllers();
+    std::vector<const Controller*> controllers() const;
```

Comment:
Can't we return something like an `std::span` or an iterator instead to avoid allocating a new vector every time this is called?

Reply:
I'm not sure how can I achieve either of your propositions because I store Controller in an `unordered_map`

Reply:
Just return a range/iterator over the pairs then